### PR TITLE
Improve hooks infos

### PR DESCRIPTION
**Pull request by richardlt**
1. Description
1. Related issues
1. About tests
1. Mentions

@ovh/cds

### CODE REVIEW

**Comment by ovh-cds** [comment]
CDS Report build-engine-cli-tools#5894.0 ✘  
* Build minimal
  * cds cli legacy ✔ 
  * cdsctl linux ✔ 
  * Engine Build linux amd64 ✔ 
  * tools/debpacker ✘  
  * tools/os-ansible-inventory ✔ 
  * Worker linux amd64 / 386 ✔

File: engine/api/router_util.go [hunk 1]
```diff
@@ -117,6 +118,59 @@ func QueryStrings(r *http.Request, key string) ([]string, error) {
...
+
+func validateSortOrder(s string) bool {
+	switch SortOrder(s) {
+	case ASC, DESC:
+		return true
+	}
+	return false
+}
+
+func SortCompareInt(i, j int, o SortOrder) bool {
```
Comment (sguiheux) [inline engine/api/router_util.go]: doc

**Comment by ovh-cds** [comment]
CDS Report ut-engine#5949.0 ✘  
* Stage 1
  * Engine Test with DB ✘

**Comment by ovh-cds** [comment]
CDS Report build-ui#5961.0 ✘  
* Compile
  * Build UI ✔ 
  * TU All -  
  * TU model ✔ 
  * TU service ✘  
  * TU shared ✔ 
  * TU views ✔ 
Unit Tests Report
* HeadlessChrome 0.0.0 (Linux 0.0.0)












* HeadlessChrome 0.0.0 (Linux 0.0.0).2016691










  * CDS: pipeline Store should add/update/delete a parameter ✘











* HeadlessChrome 0.0.0 (Linux 0.0.0).2016693


































* HeadlessChrome 0.0.0 (Linux 0.0.0).2016692

File: ui/src/app/shared/table/sorted-table.html [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    </tbody>
+    <tbody *ngIf="!loading && data.length == 0">
+        <tr>
+            <td class="center" [colSpan]="columns.length">
+                <div>{{ 'common_no_data' | translate }}</div>
+            </td>
+        </tr>
+    </tbody>
+    <tbody *ngIf="!loading && data.length > 0">
+        <tr *ngFor="let d of getDataForCurrentPage()">
```
Comment (bnjjj) [inline ui/src/app/shared/table/sorted-table.html]: try to not call function in template

File: ui/src/app/views/admin/hooks-tasks/details/hooks-task.component.ts [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+            readOnly: true
+        };
+
+        this.columns = [
+            <Column>{
+                type: ColumnType.HTML,
+                name: '',
+                selector: d => {
+                    if (d.status === HookStatus.DONE) {
+                        return '<i class="check green icon"></i>';
```
Comment (bnjjj) [inline ui/src/app/views/admin/hooks-tasks/details/hooks-task.component.ts]: I don't feel very comfortable with the fact that we write HTML in typescript. Maybe we can find a better way to do this kind of code ?

File: ui/src/app/views/admin/hooks-tasks/hooks-tasks.component.ts [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+
+    constructor(
+        private _hookService: HookService,
+        private _translate: TranslateService
+    ) {
+        this.columns = [
+            <Column>{
+                type: ColumnType.HTML,
+                name: '',
+                selector: d => d.stopped ? '<i class="stop red icon"></i>' : '<i class="play green icon"></i>',
```
Comment (bnjjj) [inline ui/src/app/views/admin/hooks-tasks/hooks-tasks.component.ts]: same here

File: ui/src/assets/i18n/fr.json [hunk 5]
```diff
@@ -389,6 +392,13 @@
   "hook_repo_man_needed": "L'application doit être liée à un dépôt",
   "hook_updated": "Hook mis à jour",
   "hook_workflow_label": "Hook : ",
+  "hook_tasks_summary": "Résumé des tâches du service Hooks",
+  "hook_task_cron": "CRON",
+  "hook_task_execs_todo": "Exécutions planifiées",
+  "hook_task_execs_total": "Total des exécutions",
+  "hook_task_repo_fullname": "Nom du repository",
```
Comment (bnjjj) [inline ui/src/assets/i18n/fr.json]: s/repository/dépôt

File: engine/hooks/hooks_handlers.go [hunk 6]
```diff
@@ -146,28 +146,62 @@ func (s *Service) postTaskHandler() service.Handler {
 	}
 }
 
+const (
+	sortKeyNbExecutionsTotal string = "nb_executions_total"
```
Comment (fsamin) [inline engine/hooks/hooks_handlers.go]: no need to specify the type

**Comment by richardlt** [comment]
I removed the HTML column type and created a Icon one. I also add optional filter to the table component and used it hook tasks page.

File: ui/src/app/shared/table/data-table.html [hunk 7]
```diff
@@ -16,17 +25,19 @@
             </td>
         </tr>
     </tbody>
-    <tbody *ngIf="!loading && data.length == 0">
+    <tbody *ngIf="!loading && dataForCurrentPage.length == 0">
```
Comment (bnjjj) [inline ui/src/app/shared/table/data-table.html]: strict equals `===`

**Pull request merged**